### PR TITLE
Add ccache (cmake) with zstd support

### PR DESCRIPTION
**Pull request by haampie**
ccache switched to cmake

### CODE REVIEW

File: var/spack/repos/builtin/packages/ccache/package.py [hunk 1]
```diff
@@ -27,9 +31,27 @@ class Ccache(AutotoolsPackage):
...
+    @when('@:3.99')
+    def build(self, spec, prefix):
+        pass
+
+    @when('@:3.99')
+    def install(self, spec, prefix):
+        configure_args = ["--prefix=" + prefix]
+        configure(*configure_args)
+        make()
+        make("install")
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/ccache/package.py:52]: Could these steps not be mapped better to be more fault tolerant?
I.e., `configure(…)` goes into `cmake`, `make()` into `build`? I've had issues with LLVM variants where the `install` step does actual work that failed, making it a lot harder to debug the underlying cause, and actually leading to an incomplete installation that is not flagged as a failed build.

Comment (haampie) [reply var/spack/repos/builtin/packages/ccache/package.py:52]: Thanks for the idea, I just did that :upside_down_face:

File: var/spack/repos/builtin/packages/ccache/package.py [hunk 2]
```diff
@@ -27,9 +31,27 @@ class Ccache(AutotoolsPackage):
...
+    depends_on('gperf', when='@:3.99')
+    depends_on('libxslt', when='@:3.99')
+    depends_on('zlib', when='@:3.99')
+
+    # Before 4.0 this was an Autotools package
+    @when('@:3.99')
+    def cmake(self, spec, prefix):
+        configure_args = ["--prefix=" + prefix]
+        configure(*configure_args)
+        pass
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/ccache/package.py]: Well, `pass` seems unnecessary now :)

File: var/spack/repos/builtin/packages/ccache/package.py [hunk 3]
```diff
@@ -7,16 +7,20 @@
 import re
 
 
-class Ccache(AutotoolsPackage):
+class Ccache(CMakePackage):
     """ccache is a compiler cache. It speeds up recompilation by caching
     previous compilations and detecting when the same compilation is being done
     again."""
 
     homepage = "https://ccache.samba.org/"
```
Comment (omor1) [inline var/spack/repos/builtin/packages/ccache/package.py]: It seems that the canonical URL is now https://ccache.dev, we might as well update to that if the package is being updated.

**Review by alalazo — CHANGES_REQUESTED** [review]
@haampie Can you take care of the homepage? Other than that LGTM

**Review by alalazo — APPROVED** [review]

**Pull request merged**